File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    let numbers = Array.from({ length: right - left + 1 }, (_, i) => left + i);
    let result = numbers.reduce((acc, num) => {
        let divisorCount = countDivisors(num);
        return divisorCount % 2 === 0 ? acc + num : acc - num;
    }, 0);
    return result;
}

function countDivisors(n) {
    let count = 0;
    for (let i = 1; i <= n; i++) {
        if (n % i === 0) count++;
    }
    return count;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    nums = []
    
    for i in range(left, right+1):
        if countDivision(i) % 2 == 0:
            nums.append(i)
        else:
            nums.append(-i)
    return sum(nums)

def countDivision(num):
    count = 0
    for i in range(1, num+1):
        if num % i == 0:
            count += 1
    return count